components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/dao/impl/ProfileDAOImpl.java

```diff
@@ -155,6 +155,29 @@ public boolean deleteProfile(Profile profile) throws ProfileManagerDAOException
     }
 
     @Override
+    public boolean deleteProfile(int profileId) throws ProfileManagerDAOException {
+        Connection conn;
+        PreparedStatement stmt = null;
+
+        try {
+            conn = this.getConnection();
+            String query = "DELETE FROM DM_PROFILE WHERE ID = ?";
+            stmt = conn.prepareStatement(query);
+            stmt.setInt(1, profileId);
+            stmt.executeUpdate();
+            return true;
+
+        } catch (SQLException e) {
+            String msg = "Error occurred while deleting the profile from the data base.";
+            log.error(msg);
```

Comment:
Fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/cdbdc86f42b350c24e9d059ff7c23e1f09242df4